### DISCUSSION TITLE
Unit Tests with TSCreateParams

### DISCUSSION THREAD

**Discussion by ascdi** [xetorthio/jedis#3943] [Q&A]
I am trying to verify that `tsAdd` was called with the correct `TSCreateParams` object.
But how can i compare one `TSCreateParams` to another?
All parameters are private, it has no `equals` implementation, nor any getters.

**Comment by sazzad16** [comment]
First of all, `tsAdd` with `TSCreateParams` is deprecated. `tsAdd` with `TSAddParams` is introduced.

Last but not least, you are welcome to craft a PR.

We have already merged PR(s) of `equals` and `hashCode` implementations. So that approach is likely to be accepted.

**Reply by ascdi** [reply]
Thanks.

thou, `TSAddParams` also has no getters nor equals :-D

If i add an equals, will i break something?

It is relatively easy to implement an equals, but maybe the code relies on checks checking for pointer (`Object.equals` Behaviour) instead of content comparison?

**Reply by sazzad16** [reply]
> If i add an equals, will i break something?

No.

> maybe the code relies on checks checking for pointer

No such stuff at this moment.

---

BTW, if you are considering to submit a PR, may I suggest to add `equals` and `hashCode` in more classes; i.e. at least all Params classes in `timeseries` package?

**Reply by ascdi** [reply]
Hi,

i can give it a try.
What branch shouldi branch off of?
There is no dev branch so i am a bit unsure

**Reply by sazzad16** [reply]
> What branch shouldi branch off of?

Usually a PR should start based from `master` branch.
More guidelines [here](https://github.com/redis/jedis/blob/master/.github/CONTRIBUTING.md).

**Reply by ascdi** [reply]
Okay, i hope i did it :-D
https://github.com/redis/jedis/pull/3959

**Reply by sazzad16** [reply]
@ascdi Thanks

**Comment by sazzad16** [comment]
Merged #3959
